### ISSUE TITLE
config settings not re-set after reconnect

### ISSUE THREAD

**Issue by ninja-** [lettuce-io/lettuce-core#598]
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/master/.github/CONTRIBUTING.md).
- [x] You specify the lettuce version and environment so it's obvious which version is affected
- [x] You provide a reproducible test case (either descriptive of as JUnit test) if it's a bug or the expected behavior differs from the actual behavior.

 
Lettuce: 4.2.1.Final (last version to have netty 4.0 support; I can't have conflicting netty versions)

So basically I am using this kind of thing

```java        
    createPublishConnection().sync().configSet("notify-keyspace-events", "AKE");
```

To get pattern subscriptions to work.
After connection drops, the channels are resubscribed and I get the `psubscribed` event - cool.
But the config is not re-set on reconnect and so keyspace notifications won't work until I put the `configSet` to `psubscribed` as a workaround.
I think that's something that lettuce should be taking care of.

**Comment by mp911de** [comment]
Lettuce 4.x supports both netty branches, 4.0 and 4.1, see the [dependency matrix chart](https://github.com/lettuce-io/lettuce-core/wiki/Dependency-Compatibility-Matrix).

I'm not sure about your proposal: There is no obvious reason why the config should be re-applied to Redis. Once you set the config values it in Redis, it's activated and `CONFIG REWRITE` persists the configuration.

On the other hand, you can't really tell what's intended by setting configuration. We can't make assumptions over the target Redis node that is chosen by a reconnect, especially in Master/Slave or Redis Cluster setups. With as little as possible assumptions, `CONFIG SET` allows you to modify your server but that's more a utility feature. You rather should configure your Redis node outside the application. I think projects providing an embedded Redis instance are good candidates for such a requirement.
